core/src/apps/common/seed.py

```diff
@@ -140,6 +140,8 @@ def derive_slip21_node_without_passphrase(path: list) -> Slip21Node:
         raise Exception("Device is not initialized")
     seed = cache.get(cache.APP_COMMON_SEED_WITHOUT_PASSPHRASE)
     if seed is None:
+        if not cache.is_session_started():
```

Comment:
This piece is interesting -- AFAICT, "node without passphrase" is not bound to a session, is that right?

I don't like that this piece of code starts its own session, and I also don't like that we cache and re-derive the same value up to 10 times.

My first instinct would be to cache it locally in seed.py, but that would mean that it doesn't get cleared on device wipe.
We could make the wipe call into seed.py to clear the cache, or we could special-case "node without passphrase" in cache (and do it the way it used to be done, with a special variable instead of a cache key).

The question is mostly whether there will ever be more data that is per-seed and not per-session.

Reply:
Yes this is something I wanted to discuss.
> AFAICT, "node without passphrase" is not bound to a session, is that right?

Good point. Maybe it indeed makes sense to have smth like `overall_cache` or `session_less_cache` (name TBD) in cache.py, which would store non-session-related items. In `cache.clear` we would clear it.

Reply:
Implemented in a4b5fdb6cea85d444d0d579fc9d20936eebf0791